runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -268,6 +260,17 @@ ILogData read(long address) {
      * @return A result, which be cached.
      */
     public Map<Long, ILogData> read(Iterable<Long> addresses) {
+        return read(addresses, true);
+    }
+
+    /**
+     * Read the given object from a range of addresses.
+     *
+     * @param addresses An iterable with addresses to read from
+     * @param waitForWrite Flag whether wait for write is required or hole fill directly.
+     * @return A map of read addresses, which will be cached if caching is enabled
+     */
+    public Map<Long, ILogData> read(Iterable<Long> addresses, boolean waitForWrite) {
```

Comment:
This is a public API now, can you document how the wait happens. Does it just wait for the greatest address in addresses and then fills the prefix without waiting ?

Reply:
Done.